client-api/src/main/java/org/corfudb/runtime/kv/core/DbVersion.java

```diff
@@ -0,0 +1,120 @@
+package org.corfudb.runtime.kv.core;
+
+import java.util.function.Supplier;
+
+/**
+ * Type contract representing the temporal uniqueness attribute of a {@link Record}.
+ */
+public interface DbVersion<T> extends Supplier<T> {
```

Comment:
I am not sure if we need this fancy a version class. It can definitely be simplified. I would rather have most things defined in protobufs.

Reply:
We have this class in protobuf, we can discuss if we should simplify Version class